analytics/views.py

```diff
@@ -1,31 +1,107 @@
-from __future__ import absolute_import
-from __future__ import division
-from typing import Any, Dict, List, Tuple, Optional, Sequence, Callable, Union, Text
+from __future__ import absolute_import, division
 
+from django.core import urlresolvers
 from django.db import connection
 from django.db.models.query import QuerySet
-from django.template import RequestContext, loader
-from django.core import urlresolvers
 from django.http import HttpResponseNotFound, HttpRequest, HttpResponse
+from django.template import RequestContext, loader
+from django.utils import timezone
+from django.utils.translation import ugettext as _
 from jinja2 import Markup as mark_safe
 
-from zerver.decorator import has_request_variables, REQ, zulip_internal
-from zerver.models import get_realm, UserActivity, UserActivityInterval, Realm
-from zerver.lib.timestamp import timestamp_to_datetime
+from analytics.lib.counts import CountStat, process_count_stat, COUNT_STATS
+from analytics.models import RealmCount, UserCount
+
+from zerver.decorator import has_request_variables, REQ, zulip_internal, \
+    zulip_login_required, to_non_negative_int, to_utc_datetime
+from zerver.lib.request import JsonableError
+from zerver.lib.response import json_success
+from zerver.lib.timestamp import ceiling_to_hour, ceiling_to_day, timestamp_to_datetime
+from zerver.models import Realm, UserProfile, UserActivity, UserActivityInterval, \
+    get_realm
+from zproject.jinja2 import render_to_response
 
 from collections import defaultdict
 from datetime import datetime, timedelta
 import itertools
-import time
-import re
+import json
 import pytz
-from six.moves import filter
-from six.moves import map
-from six.moves import range
-from six.moves import zip
-eastern_tz = pytz.timezone('US/Eastern')
+import re
+import time
 
-from zproject.jinja2 import render_to_response
+from six.moves import filter, map, range, zip
+from typing import Any, Dict, List, Tuple, Optional, Sequence, Callable, Union, Text
+
+@zulip_login_required
+def stats(request):
+    # type: (HttpRequest) -> HttpResponse
+    return render_to_response('analytics/stats.html', request=request)
```

Comment:
I feel like I need to pass the request object here so that we can get the user_profile in get_chart_data below. But how is the information being passed from here to there? (Or said differently, how does `@has_request_variables` on the line below know the user_profile?)